util/src/main/java/tech/pegasys/teku/util/async/SafeFuture.java

```diff
@@ -22,18 +22,16 @@
 import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Stream;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 
 public class SafeFuture<T> extends CompletableFuture<T> {
-  private static final Logger LOG = LogManager.getLogger();
 
   public static SafeFuture<Void> COMPLETE = SafeFuture.completedFuture(null);
 
   public static void reportExceptions(final CompletionStage<?> future) {
     future.exceptionally(
         error -> {
-          LOG.error("Unhandled exception", error);
+          final Thread currentThread = Thread.currentThread();
+          currentThread.getUncaughtExceptionHandler().uncaughtException(currentThread, error);
```

Comment:
This seems like a good idea - but I'm afraid we'll end up bubbling up a lot of noise to the status logger now ...  Maybe the right answer is to make this change and add more custom handling where too much noise bubbles up.

Reply:
Yeah, we've have a lot of people confused that Teku was "doing nothing" because they couldn't see the errors on the console but in the logs errors were appearing all over the place.  Adding all the needed error handling instead of letting things reach the uncaught exception level (which is what `reportExceptions` is meant to be (possibly need a better name) is good.  We potentially should also set things up so that the console log doesn't show stack traces but the log file does.